server/inspect.go

```diff
@@ -78,9 +78,24 @@ func (s *Server) getContainerInfo(id string, getContainerFunc, getInfraContainer
 			image = status.Name
 		}
 	}
+
+	pidToReturn := ctrState.Pid
+	if ctr.Spoofed() && pidToReturn == 0 {
+		// This is possible when we've dropped the infra container
+		// and this request is for a pod.
+		// Since the Pid is used exclusively to find the network stats,
+		// and pods share their network (whether it's host or pod level)
+		// we can return the pid of a running container in the pod.
+		for _, c := range sb.Containers().List() {
```

Comment:
Can we audit and make sure that the pid isn't used for anything else on the cadvisor side? :)

Reply:
it is also used for printing the container:
`klog.V(4).Infof("Could not get PIDs for container %d: %v", h.pid, err)`

but the only functional uses seem to be network related